R/eda.R

```diff
@@ -79,8 +79,43 @@ describe_cat_var <- function(dataframe, cat_vars) {
 #' df <- data.frame(x = (c(2,3,4)), y= c(1,10,3))
 #' col_num <- list("x", "y")
 #' calc_cor(df, col_num)
-calc_cor <- function(dataframe, num_vars) {
-  #TODO implement function
+calc_cor <- function(df, num_vars) {
+  # Test dataframe input to check if dataframe is a dataframe
+  if (!is.data.frame(df))
+    stop("Input 'df' should be a dataframe.")
+
+  # Find numerical columns and remove NA
+  df_num <- df %>%
+    select(num_vars) %>%
+    drop_na()
+
+  # Test colums to check if columns provided are numeric
+  for (i in num_var) {
+    if (typeof(df_num[,i]) == "list") {
+      data_unlisted <- unlist(df_num[i])
+      if (!is.numeric(data_unlisted))
+        stop("Columns do not all contain numeric values.")
+    } else if (!is.numeric(df_num[, i])) {
+      stop("Columns do not all contain numeric values.")
+    }
+  }
```

Comment:
Does the `num_var` only take list as input? I thought in R a vector of strings is more common.

Reply:
No, it should take a tibble or dataframe. I was having trouble with tibbles, so I put in the condition if it was a list. I could also change it the way you did to make tibbles into dataframes.

Reply:
No, I did't make `tibble` into `data.frame`, but if the input is `data.frame`, I make it into `tibble`. 
Sorry I misread the purpose of this chunk of code before. If you only want to check if only numeric columns are selected, `all(sapply(df_num, is.numeric))` should work.

Reply:
That's much cleaner! Thank you! I will make that change.

Reply:
Added revised code in PR #36